stdlib/contrib/RohanSreerama5/naiveBayesClassifier/naiveBayesClassifier.flux

```diff
@@ -0,0 +1,114 @@
+//Final working code as of August 11, 2020
+//Currently supports single field classification and binary data sets 
+
+package naiveBayesClassifier
+
+import "system"
+
+
+
+////////csv.from(csv: naiveBayesClassifier/zoo_data.csv) |> naiveBayes()
+
+//---
+naiveBayes = (tables=<-, myClass, myField, myMeasurement) => {
+
+//naiveBayes = (myClass,  myField, myMeasurement) => {
+//r = tables 
+training_data = tables
+  //from(bucket: myBucket)
+  |> range(start:2020-01-02T00:00:00Z, stop: 2020-01-06T23:00:00Z) //data for 3 days 
+  |> filter(fn: (r) => r["_measurement"] == myMeasurement and r["_field"] == myField)
+  |> group()
+  //|> yield(name: "trainingData")
+
+test_data = tables
+  //from(bucket: myBucket)
+  |> range(start:2020-01-01T00:00:00Z, stop: 2020-01-01T23:00:00Z) //data for 1 day
+  |> filter(fn: (r) => r["_measurement"] == myMeasurement and r["_field"] == myField)
+  |> group()
+  //|> yield(name: "test data")
+
+//data preparation 
```

Comment:
I think there are a few ways to make this script reusable so that it can work with more than just the one data set in the future. If you maybe pulled the data prep out of this function and made it its own module and then standardized column naming, then others could use your functions as well. We can discuss this more if you're interested in opening another PR to further modify this. 

For now, I think this is a good guideline for folks who want to use Flux for these types of algorithms in the future.

Reply:
Hi Faith,
Just wanted to let you know after some messing around, I got the bayes classifier to finally run make generate successfully. Only things different I've done is install ruby. I'm also not sure if this is related or not, but as a side thing, I cloned down a fresh copy of the flux repo master and when I ran make, it did not make and complained about the '/r' line break character. So I installed a Unix program called dos2unix and ran dos2unix pkg-config.sh which is supposed to convert the line breaks in the text file to Unix format. Once I did this and ran make again, the flux repo ran make successfully. Anyways, my project is now compiling properly on my machine so if you can approve of this, I'll do some code cleanup very soon and we can push this to master! Please get back to me when you get a chance.

Reply:
Yeah, generally things look good now. You do need to pass all the CI checks in order for this to be merged. See the notes I left below.

Reply:
I just ran go mod tidy and make test. I think I have another rust issue. This time, I think I need rustup installed, but when I try to install rustup, it panics. I'm searching the web for possible solutions but I'm not too sure where to go with this. Do you have a few mins to spare on Zoom?

Reply:
I might have some time later. What is the panic that you're seeing?

Reply:
It is panicking when I run this to install rustup. 
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `22`,
 right: `4`', src/libstd/sys/unix/thread.rs:179:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `22`,
 right: `4`', src/libstd/sys/unix/thread.rs:179:21
stack backtrace:
   0:     0x7f7bf9edaffd - backtrace::backtrace::libunwind::trace::h812748238d609e46
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1:     0x7f7bf9edaffd - backtrace::backtrace::trace_unsynchronized::h7c97e818aebf09c8
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2:     0x7f7bf9edaffd - std::sys_common::backtrace::_print_fmt::h60d914263b0ccd71
                               at src/libstd/sys_common/backtrace.rs:78
   3:     0x7f7bf9edaffd - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf78227137afc7565
                               at src/libstd/sys_common/backtrace.rs:59
   4:     0x7f7bf9b6524c - core::fmt::write::h543cdf60775f89bf
                               at src/libcore/fmt/mod.rs:1069
   5:     0x7f7bf9eda904 - std::io::Write::write_fmt::h0c7f3ce24c679426
                               at src/libstd/io/mod.rs:1504
   6:     0x7f7bf9eda105 - std::sys_common::backtrace::_print::h80e55e24be231368
                               at src/libstd/sys_common/backtrace.rs:62
   7:     0x7f7bf9eda105 - std::sys_common::backtrace::print::h3b197b9c1261c865
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x7f7bf9eda105 - std::panicking::default_hook::{{closure}}::ha6c807149ce20f8f
                               at src/libstd/panicking.rs:198
   9:     0x7f7bf9ed98f4 - std::panicking::default_hook::he49a9c12e358cc45
                               at src/libstd/panicking.rs:218
  10:     0x7f7bf9ed93b6 - std::panicking::rust_panic_with_hook::h93f74f5ef2f71f31
                               at src/libstd/panicking.rs:515
  11:     0x7f7bf9ed9198 - rust_begin_unwind
                               at src/libstd/panicking.rs:419
  12:     0x7f7bf9ed9140 - std::panicking::begin_panic_fmt::hfa6ef29ba81f400e
                               at src/libstd/panicking.rs:373
  13:     0x7f7bf9e11671 - <rustup::diskio::threaded::Threaded as rustup::diskio::Executor>::join::habf6ee901a47a8a4
  14:     0x7f7bf9e10c78 - core::ptr::drop_in_place::h0b1b0dcb4b07c267
  15:     0x7f7bf9d76c40 - core::ptr::drop_in_place::h09572fbec04e583f
  16:     0x7f7bf9e2842e - rustup::dist::component::package::unpack_without_first_dir::h99f4b0c2cae349ff
  17:     0x7f7bf9de559f - rustup::dist::manifestation::Manifestation::update::hacf67fad3b44f03a
  18:     0x7f7bf9dd563c - rustup::dist::dist::update_from_dist_::h1462627f9c58c494
  19:     0x7f7bf9dd227a - rustup::install::InstallMethod::install::hbb6430b2dfaf5792
  20:     0x7f7bf9dd0d21 - rustup::toolchain::DistributableToolchain::install_from_dist::h7b9bf9317c9aa2c5
  21:     0x7f7bf9ec5aa2 - rustup::cli::self_update::install::h8ca0f61a773526a9
  22:     0x7f7bf9ecbd3d - rustup::cli::setup_mode::main::h4d335e1871435d70
  23:     0x7f7bf9adf42a - rustup_init::main::h22da971c634937a2
  24:     0x7f7bf9ef3e83 - std::rt::lang_start_internal::{{closure}}::{{closure}}::h4ed4ab1fb893cc93
                               at src/libstd/rt.rs:52
  25:     0x7f7bf9ef3e83 - std::sys_common::backtrace::__rust_begin_short_backtrace::h1f01c818c00c4f70
                               at src/libstd/sys_common/backtrace.rs:130
  26:     0x7f7bf9ae053f - main
  27:     0x7f7bf96270b3 - __libc_start_main
  28:     0x7f7bf9adb029 - <unknown>
thread panicked while panicking. aborting.
Illegal instruction (core dumped)

Reply:
Basically, make test is complaining that I don't have a default Rust toolchain set

Reply:
I think you mentioned that you're now working on a Windows device. You may still be using the Mac installation method.  I think that the Windows method should be accessible using this webpage: https://rustup.rs

Reply:
Yeah, that website is actually where I copied the curl command to install rust.

Reply:
if you look at the website, it says not to use the curl command for Windows. There an exe file that you download instead.

Reply:
O actually, I'm on Windows Subsystem for Linux. So it would make sense to use that. Sorry, should've been more clear.

Reply:
Yeah, I'm not an expert on running Windows w/ Linux, but I would try the Windows instructions.

Reply:
Or if that can't be done, then the Windows/Linux clash might be part of the issue here. I think I'd be out of my depth if that's the case.

Reply:
I've been using it for a while for school stuff too and its basically just a Linux machine that I'm running. So I'm not entirely sure its an OS problem. I'll keep looking into it though -thx!

Reply:
hi. Its the end of my day here, but this guide should set you on the right course. https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

First, you'll have a merge conflict with `stdlib/test_packages.go`. When you try to rebase it will ask you to fix that. 

Then, start following the guide. You can do: 
- `git rebase -i HEAD~9`, then you should see a list of all your commits with the `pick` command there. If you're not set up with interactive rebase you'll be in vim and you can press `i` to start editing. 
- Replace all the `pick` with `squash` except the first one. This will squash all your subsequent commits go into the first one. Replace the first `pick` with `reword` while one that same screen. 
- YOu'll have another prompt to reword the commit. Follow the semantic commit message convention by writing a commit message that completes the sentence: "This commit will...."

Give that a try. If you're having trouble ping me here early if your day and I'll see if there's time on my calendar to pair.

Reply:
Example of how I'd reword your first commit: 
`feat(stdlib/contrib): add support for single field classification using naive bayes`

Reply:
Hi, I was able to squash the commits via rebase locally. But I'm a bit unsure about how to make this change show up on the Git remote branch

Reply:
Hey rohan. I'm not at my computer right now, but if you type the command  `git push --force`, it should override the remote commits and replace them with your local ones. I'd double and triple check that your local branch is correct first by checking `git status` and then making sure all your changes are up to date.

Reply:
Yeah. Now it's complaining about my PR title, which is odd because I did use the word 'feat' in it. Also, before I forget, I wanted to mention something about the code. On lines 43 and 81 of naiveBayesClassifier.flux, there exists some code that is specific to the Animal dataset that I had used. Specifically on line 43 where I do `r.airborne`, last I spoke with Anais, she told me that in Flux there isn't a way to turn that into a variable and do something like `r.{myField}`.  I think for the time being, we can push this up to master and address these nits soon after if this sounds alright to you.

Reply:
yeah, I added a merge commit because you still had a merge conflict on this branch that you hadn't fixed yet. I fixed it for you, but I can't update the commit message since its not my branch. It will just have to fail that check for now.

Reply:
Yeah there's a lot of stuff in this code that is specific to your example and may not work modularly with other people's datasets. I think its okay to revisit it when you have time since its in a contrib folder.

Reply:
@RohanSreerama5 i just got back. I've got a few minutes if you'd like to resolve the test failure.

Reply:
Sure I'm game. I'll send you a zoom link via email